demonstrations/tutorial_vqt.py

```diff
@@ -0,0 +1,826 @@
+given by:
+
+.. math::
+    \mathcal{L}(\theta, \ \phi) \ = \ \beta \langle \hat{H} \rangle \ - \ S_\theta \ = \
+    \beta \ \text{Tr} (\hat{H} \ \rho_{\theta \phi}) \ - \ S_\theta \ = \ \beta \ \text{Tr}( \hat{H} \ \hat{U}(\phi)
+    \rho_{\theta} \hat{U}(\phi)^{\dagger} ) \ - \ S_\theta,
+
+where :math:`\rho_\theta` is the initial density matrix, :math:`U(\phi)`
+is the paramterized ansatz, and :math:`S_\theta` is the von Neumann
+entropy of :math:`\rho_{\theta \phi}`. It is important to note that the
+von Neumann entropy of :math:`\rho_{\theta \phi}` is the same as the von
+Neumann entropy of :math:`\phi_{\theta}`, since entropy is invariant
+under unitary transformations:
+
+.. math::
+    S(\rho') = \ - \text{Tr} (\rho' \log \rho') &= \ - \text{Tr} ( U \rho U^{\dagger} \log (U \rho U^{\dagger}))  \\
+             &= - \text{Tr} ( U \rho U^{\dagger} \log \rho)  \\
+             &= \ - \text{Tr} ( U \rho \log \rho U^{\dagger}) \\
+             &= - \text{Tr} ( \rho \log \rho U^{\dagger} U) \\
+             &= \ - \text{Tr} ( \rho \log \rho) \\
+             &= \ S(\rho)
+
+We repeat the algorithm with new parameters until we minimize free
+energy. Once we have done this, we have arrived at the thermal state.
+This comes from the fact that our free energy cost function is equivalent
+to the relative entropy between :math:`\rho_{\theta \phi}` and our
+target thermal state. Relative entropy is defined as:
+
+.. math:: D(\rho_1 || \rho_2) \ = \ \text{Tr} (\rho_1 \log \rho_1) \ - \ \text{Tr}(\rho_1 \log \rho_2)
+
+If we let :math:`\rho_1` be :math:`\rho_{\theta \phi}` and
+:math:`\rho_2` be our thermal state, we get:
+
+.. math::
+    D(\rho_{\theta \phi} || \rho_{\text{Thermal}}) &= -S_{\theta} \ - \ \text{Tr}(\rho_{\theta \phi} (-\beta \hat{H} \ - \ \log Z_{\beta})) \\
+    &= \beta \text{Tr} (\rho_{\theta \phi} \hat{H}) \ + \ \log Z_{\beta} \text{Tr}(\rho_{\theta \phi}) \ - \ S_{\theta} \\
+    &= \beta \langle \hat{H} \rangle \ - \ S_{\theta} \ + \ \log Z_{\beta} \\
+    &= \mathcal{L}(\theta, \ \phi) \ + \ \log Z_{\beta}
+
+Since relative entropy must be positive, and is clearly :math:`0` when
+:math:`\rho_{\theta \phi} \ = \ \rho_{\text{Thermal}}`, it follows that
+relative entropy, and hence :math:`\mathcal{L}(\theta, \ \phi)` (since
+it only differs from rleative entropy by an overall additive constant),
+are minimized when
+:math:`\rho_{\theta \phi} \ = \ \rho_{\text{Thermal}}`. So, we know that
+we have to minimize :math:`\mathcal{L}` to find the thermal state. More
+specifically, when
+:math:`\mathcal{L}(\theta, \ \phi) \ = \ - \log Z_{\beta}`, then we have
+minimized the cost function and have found the thermal state.
+
+For a diagramatic representation of how this works, check out Figure 3
+from the `original VQT paper <https://arxiv.org/abs/1910.02071>`__.
+
+The 3-Qubit Ising Model on a Line
+---------------------------------
+
+We will begin by consdering the Ising model on a linear graph, for 3
```

Comment:
```suggestion
We begin by consdering the Ising model on a linear graph, for three
```
It's better style to use words (e.g. two, seven) to denote numbers smaller than ten

Reply:
Also, why just three qubits? I guess it's too late, but I would assume that we could push qubit number a bit further, no?

Reply:
I agree, 3 qubits isn't a lot, however, I didn't want the simulations to require a ton of steps, and thus take a long time to converge (this tutorial is already pushing the 10 minute cap on runtime).

Reply:
It eventually does a simulation for the case of 4 quits as well